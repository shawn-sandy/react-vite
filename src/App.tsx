import React, { useState } from 'react'
import './App.css'

// import {Button} from '@fpkit/fp-button'
import {Badge, FP, Dropdown, Card, Button, Kit, Box } from '@fpkit/fp'

function App() {
  const [count, setCount] = useState(0)

  return (
    <div className="App">
      <header className="App-header">
        <p>Hello Vite + React!</p>
        <p>
          <button type="button" onClick={() => setCount((count) => count + 1)}>
            count is: {count}
          </button>
        </p>
        <p>
          Edit <code>App.tsx</code> and save to test HMR updates.
        </p>
        <p>
          <a
            className="App-link"
            href="https://reactjs.org"
            target="_blank"
            rel="noopener noreferrer"
          >
            Learn React
          </a>
          {' | '}
          <a
            className="App-link"
            href="https://vitejs.dev/guide/features.html"
            target="_blank"
            rel="noopener noreferrer"
          >
            Vite Doc </a>
        </p>
        <Button type="button" onClick={() => setCount((count) => count + 1)}>Increment Count</Button>

        <p>
        <FP as='h3'>FP KIt <Badge elm="span" role="note">Beta</Badge></FP>
        </p>

        <p>
        <Kit as='h3'>FP KIt <Badge elm="span" role="note">Kit Beta</Badge></Kit>
        </p>

        <p>
        <Box as='h3'>FP KIt <Badge elm="span" role="note">Beta</Badge></Box>
        </p>
        <Card elm="aside">
          <p>Lorem ipsum dolor sit amet consectetur adipisicing elit. Quisquam iusto suscipit ea vitae, accusamus ipsum. Nobis nihil deserunt enim debitis laboriosam suscipit, eius impedit dicta ex animi. Quos, voluptatum assumenda?</p>
        </Card>
        <Dropdown summary='Dropdown' title='' styl={{backgroundColor: "black", textAlign: "left"}}>
          <p>
            Lorem ipsum dolor sit amet, consectetur adipisicing elit. Rerum unde ex eos cum magnam. Et veniam laudantium commodi similique, temporibus beatae porro. Eum, recusandae. Minus voluptatum quisquam consequatur voluptate est! <Badge role='note'>Beta</Badge>
          </p>
          </Dropdown>
          <p>
          <Card>
            Lorem, ipsum dolor sit amet consectetur adipisicing elit. Laudantium quibusdam quo dignissimos, error incidunt tempore facere illo dolore dicta eius dolorum nesciunt, facilis obcaecati. Placeat sunt dolor eveniet explicabo minima.
          </Card>
          </p> 
      </header>
    </div>
  )
}

export default App
