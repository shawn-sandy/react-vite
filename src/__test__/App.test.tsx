import React from 'react';

import { describe, expect, it, test } from 'vitest';
import { render, screen } from '@testing-library/react';

import App from '../App'

describe('<App>', () => {

  test('App component should render correctly', () => {
    const Component = render(<App/>)


    const logo = screen.getByRole('img', {  name: /logo/i});
    screen.debug(logo)

    expect(Component).toMatchSnapshot();
    expect(Component.getAllByRole('link', { name: /learn react/i})).toBeDefined()
    expect(Component.getByRole('link', { name: /vite docs/i})).toBeDefined()
    expect(Component.getByRole('button', {  name: /count is: 0/i})).toBeDefined()
  })
})
