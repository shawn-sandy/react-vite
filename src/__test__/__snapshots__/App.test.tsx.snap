// Vitest Snapshot v1

exports[`<App> > App component should render correctly 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="App"
      >
        <header
          class="App-header"
        >
          <img
            alt="logo"
            class="App-logo"
            src="/src/logo.svg"
          />
          <p>
            Hello Vite + React!
          </p>
          <p>
            <button
              type="button"
            >
              count is: 
              0
            </button>
          </p>
          <p>
            Edit 
            <code>
              App.tsx
            </code>
             and save to test HMR updates.
          </p>
          <p>
            <a
              class="App-link"
              href="https://reactjs.org"
              rel="noopener noreferrer"
              target="_blank"
            >
              Learn React
            </a>
             | 
            <a
              class="App-link"
              href="https://vitejs.dev/guide/features.html"
              rel="noopener noreferrer"
              target="_blank"
            >
              Vite Docs
            </a>
          </p>
        </header>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="App"
    >
      <header
        class="App-header"
      >
        <img
          alt="logo"
          class="App-logo"
          src="/src/logo.svg"
        />
        <p>
          Hello Vite + React!
        </p>
        <p>
          <button
            type="button"
          >
            count is: 
            0
          </button>
        </p>
        <p>
          Edit 
          <code>
            App.tsx
          </code>
           and save to test HMR updates.
        </p>
        <p>
          <a
            class="App-link"
            href="https://reactjs.org"
            rel="noopener noreferrer"
            target="_blank"
          >
            Learn React
          </a>
           | 
          <a
            class="App-link"
            href="https://vitejs.dev/guide/features.html"
            rel="noopener noreferrer"
            target="_blank"
          >
            Vite Docs
          </a>
        </p>
      </header>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
